pkg/dns/aws/filewatcher.go

```diff
@@ -0,0 +1,76 @@
+package aws
+
+import (
+	"bytes"
+	"io/ioutil"
+	"time"
+
+	certsutil "github.com/openshift/cluster-ingress-operator/pkg/util/certs"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// startWatcher starts the FileWatcher and periodically ensures DNS clients
+// are using the current ca bundle until a message is received on the stop
+// or error channels.
+func (m *Provider) startWatcher(stop <-chan struct{}) error {
+	go wait.Until(func() {
+		if err := m.ensureDNSClientsTLSTransport(); err != nil {
+			log.Error(err, "failed to ensure dns client tls transport")
+		}
+	}, 1*time.Minute, stop)
```

Comment:
Is this a second, polling-based watcher built on top of the fsnotify-based watcher?  This is very confusing.

Reply:
@Miciah the fsnotify-based watcher is continuously watching `fileName` and updating `currentData`.  `startWatcher()` starts the fsnotify-based watcher and starts a goroutine that periodically ensures dns clients are using the latest ca bundle by comparing the fsnotify-based watcher's `currentData` with the data from reading `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`.  Let me know if you have any suggestions to improve. Maybe have `handleEvent()` to update the dns clients?

Reply:
What I find confusing is that you essentially have two watchers: a polling-based watcher and an fsnotify-based watcher, where the former is built atop the latter, and half the watcher logic is in the aws provider package and half is in the watcher package.  Is the polling-based watcher logic going to be repeated in every provider implementation?

I would instead amend the fsnotify-based watcher to watch two channels: the fsnotify events channel and a ticker channel.  Dan did something similar in https://github.com/openshift/router/pull/61/commits/4a4f5ab03fcc31d9260d892863dc85ec0ae6a924 (that commit has some extra logic to distinguish between reloads triggered by fsnotify events and reloads triggered by ticks, but otherwise it does essentially what I am am suggesting).  Although others may disagree, I find the single combined fsnotify- and ticker-based watcher easier to understand and reason about.

Reply:
@Miciah commit `4c6eb35` updated the PR and my testing against a live cluster produced expected results. However, the commit did not use a bool chan to update elb, tagging and route53 sessions. Commit `97eb21d` implements the chan approach that you suggest. I'm waiting for my cluster to finish installing to test. I'll start testing this commit tomorrow morning PT. Thanks for the guidance!